File: CalculatorProject/src/calculator.py
from calculator_exception import CalculatorException

class Calculator:
    def perform_operation(self, operation, num1, num2):
        if operation == "add":
            return self.add(num1, num2)
        elif operation == "subtract":
            return self.subtract(num1, num2)
        elif operation == "multiply":
            return self.multiply(num1, num2)
        elif operation == "divide":
            return self.divide(num1, num2)
        else:
            raise CalculatorException("Invalid operation.")

    def add(self, num1, num2):
        return num1 + num2

    def subtract(self, num1, num2):
        return num1 - num2

    def multiply(self, num1, num2):
        return num1 * num2

    def divide(self, num1, num2):
        if num2 == 0:
            raise CalculatorException("Cannot divide by zero.")
        return num1 / num2

File: CalculatorProject/src/Calculator.java
public class Calculator {
    public double performOperation(String operation, double num1, double num2) throws CalculatorException {
        switch (operation) {
            case "add":
                return add(num1, num2);
            case "subtract":
                return subtract(num1, num2);
            case "multiply":
                return multiply(num1, num2);
            case "divide":
                return divide(num1, num2);
            default:
                throw new CalculatorException("Invalid operation.");
        }
    }

    private double add(double num1, double num2) {
        return num1 + num2;
    }

    private double subtract(double num1, double num2) {
        return num1 - num2;
    }

    private double multiply(double num1, double num2) {
        return num1 * num2;
    }

    private double divide(double num1, double num2) throws CalculatorException {
        if (num2 == 0) {
            throw new CalculatorException("Cannot divide by zero.");
        }
        return num1 / num2;
    }
}
